crates/runtime/src/embeddings/connector.rs

```diff
@@ -262,24 +273,34 @@ impl DataConnector for EmbeddingConnector {
                 .boxed();
 
             return Some(stream);
-        }
 
-        let embedding_table = Arc::new(
-            table_provider
-                .as_any()
-                .downcast_ref::<EmbeddingTable>()?
-                .clone(),
-        );
-        let underlying_table = Arc::clone(&embedding_table.base_table);
-        let underlying_federated_table = Arc::new(FederatedTable::Immediate(underlying_table));
-
-        let stream = self
-            .inner_connector
-            .changes_stream(underlying_federated_table)?
-            .then(move |item| Self::embed_change_envelope(item, Arc::clone(&embedding_table)))
-            .boxed();
-
-        Some(stream)
+        // `VectorScanTableProvider` is generally wrapped by a `IndexedTableProvider` (as above), but in the case both [`Self`] and the [`FullTextConnector`] exist, the latter will unwrap the `IndexedTableProvider` first. It will correctly handle indexing vector indexes as that point.
+        } else if let Some(vector_scan) = table_provider
```

Comment:
This logic is pretty ugly - not your fault, its a consequence of the current design. I think its fine for now, but we should have a think about what a better way to do this would be.

Reply:
I agree, this is partially true with the `EmbeddingTable` too